packages/espressocash_app/lib/core/coingecko_client.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:dio/dio.dart';
+import 'package:dio_cache_interceptor/dio_cache_interceptor.dart';
+import 'package:dio_cache_interceptor_db_store/dio_cache_interceptor_db_store.dart';
+import 'package:path_provider/path_provider.dart';
+
+class CoingeckoClient {
+  CoingeckoClient() {
+    init();
+  }
+
+  late final Dio dio;
+  late final CacheOptions options;
+
+  Future<void> init() async {
+    dio = Dio();
+
+    final directory = await getTemporaryDirectory();
+
+    options = CacheOptions(
+      store: DbCacheStore(databasePath: directory.path),
+      policy: CachePolicy.refreshForceCache,
+    );
+
+    dio.interceptors.add(
+      DioCacheInterceptor(options: options),
+    );
+  }
+
+  Dio setMaxAge(Duration maxAge) => dio
+    ..interceptors
+        .insert(0, CacheInterceptor(options: options, maxAge: maxAge));
+}
+
+class CacheInterceptor extends Interceptor {
```

Comment:
Why not built-in `DioCacheInterceptor`?

Reply:
It uses both `DioCacheInterceptor` and a custom `CacheInterceptor`. The library uses the response's cache headers to decide how to long cache. Since we do not have control over the server; and our use case would be to cache the data for custom x amount of time (ie cache details api for 6 hrs), a custom interceptor to handle max age is needed.

ref: https://github.com/llfbandit/dio_cache_interceptor/issues/6